cylc/flow/id_cli.py

```diff
@@ -490,21 +491,46 @@ async def _expand_workflow_tokens_impl(tokens, match_active=True):
 
 
 def _parse_src_path(id_):
+    """Parse CLI workflow arg to find a valid source directory.
+
+    Returns:
+      - (dir name, dir path, config file path) if id_ is a valid src dir.
+      - or None, if id_ could be a workflow ID
+
+    A valid source directory is:
+      - an existing directory that contains a worklow config file
+    and not a relative path (which could be a workflow ID), i.e. it must be:
+      - the current directory (".")
+      - or a directory path that starts with "./"
+      - or an absolute directory path
+
+    It's OK if id_ happens to match a relative path to an existing directory or
+    file (other than a workflow config file) because there could be a workflow
+    ID with the same name.
+
+    """
+    abort_if_flow_file_in_path(Path(id_))
     src_path = Path(id_)
     if (
-        id_ == os.curdir
-        or id_.startswith(f'{os.curdir}{os.sep}')
-        or Path(id_).is_absolute()
+        id_ != os.curdir
+        and not EXPLICIT_RELATIVE_PATH_REGEX.match(id_)
```

Comment:
the regex will match `.` alone so that isn't needed
```suggestion
        not EXPLICIT_RELATIVE_PATH_REGEX.match(id_)
```

Reply:
Oops yes.